proto/ibc/core/client/v2/counterparty.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package ibc.core.client.v2;
+
+option go_package = "github.com/cosmos/ibc-go/v9/modules/core/02-client/types";
+
+import "cosmos/msg/v1/msg.proto";
+import "gogoproto/gogo.proto";
+
+// CounterpartyInfo defines the key that the counterparty will use to message our client
+message CounterpartyInfo {
+  // merkle prefix key is the prefix that ics provable keys are stored under
+  repeated bytes merkle_prefix = 1;
+  // client identifier is the identifier used to send packet messages to our client
+  string client_id = 2;
+}
+
+// CounterpartyMsg defines the ibc/client CounterpartyMsg service.
+service CounterpartyMsg {
+  option (cosmos.msg.v1.service) = true;
+
+  // RegisterCounterparty defines a rpc handler method for MsgRegisterCounterparty.
+  rpc RegisterCounterparty(MsgRegisterCounterparty) returns (MsgRegisterCounterpartyResponse);
+}
+
+// MsgRegisterCounterparty defines a message to register a counterparty on a client
+message MsgRegisterCounterparty {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  option (gogoproto.goproto_getters) = false;
+
+  // client identifier
+  string client_id = 1;
+  // counterparty merkle prefix
+  repeated bytes merkle_prefix = 2;
```

Comment:
To avoid confusion, I would also name this counterparty_merkle_prefix.
Or restructure the message like this:
```proto
string client_id = 1;
CounterpartyInfo counterparty = 2;